src/lib/FreeC/Backend/Coq/Analysis/ConstantArguments.hs

```diff
@@ -143,8 +143,9 @@ makeConstArgGraph decls = do
           checkExpr e1 [] && checkExpr e2 [] && checkExpr e3 []
         checkExpr (IR.Case _ expr alts _) _ =
           checkExpr expr [] && all (flip checkAlt []) alts
-        checkExpr (IR.Let _ binds expr _) _ =
-          checkExpr expr [] && all (flip checkBind []) binds
+        checkExpr (IR.Let _ binds expr _) _
+          | x `shadowedBy` map IR.bindVarPat binds = not (callsG expr)
+          | otherwise                              = checkExpr expr []
```

Comment:
The right-hand sides of the bindings still need to be checked with `checkExpr` and `callsG`, respectively.